.travis.yml

```diff
@@ -42,10 +42,9 @@ jobs:
       script:
         - yarn tx
         - yarn build:$COZY_APP_SLUG:browser
-        - ./scripts/choose-test-instance.sh
-        - source setinstance.sh
         - &downcloud-cert if [ "$TRAVIS_SECURE_ENV_VARS" != "false" ]; then eval "$(ssh-agent -s)" && chmod 600 /tmp/id_rsa_travis_downcloud && ssh-add /tmp/id_rsa_travis_downcloud; fi # Needs to be inlined and not in a script to work.
-        - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then APP_URL=$(node scripts/downcloud.js build/$COZY_APP_SLUG) && ./scripts/deploy-and-test-on-instance.sh $APP_URL; fi
+        - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then ./scripts/create-instance-and-test.sh; fi
+        - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then ./scripts/delete-instance.sh; fi
```

Comment:
I can merge `create instance` , `deploy instance` and `test` in one file as the execution of `tests` depends on `deploy instance`, which depend on `create instance`, but I think we should keep the `delete instance` apart.
If the `create-deploy-test` script fail at some point, we still want to try to `delete the instance` .
Otherwise, if we restart the buid (or commit more stuff to the PR), there is a risk that the instance already exist, so the build might fail again.

Reply:
> If the create-deploy-test script fail at some point, we still want to try to delete the instance .

Yep ok, that makes sense. I think it's still worth splitting or merging the other ones 👍

> Otherwise, if we restart the buid (or commit more stuff to the PR), there is a risk that the instance already exist, so the build might fail again.

I guess that's a risk anyway, I think travis jobs can be interrupted at pretty much any point, either manually or by other commits. But you're right, lets keep that risk as low as possible.